psakefile.ps1

```diff
@@ -0,0 +1,55 @@
+include "./psake-build-helpers.ps1"
+
+$tag = $(git tag -l --points-at HEAD)
+$revision = @{ $true = ""; $false = "local" }[$env:BUILD_BUILDNUMBER -ne $NULL]
+$suffix = @{ $true = ""; $false = "ci-$revision" }[$tag -ne $NULL -and $revision -ne "local"]
+
+properties {
+	$configuration = 'Release'
+	$projectRoot = Resolve-Path "./"
+	$buildDir = "$projectRoot/Build"
+	$testResultsDir = "$projectRoot/src/BulkWriter.Tests/TestResults"
+	$versionPrefix = "3.0.0"
+	$version = "ci-local"
+}
+
+task default -depends Run-Tests
+task Run-CI -depends Clean-Solution, Run-Tests -description "Continuous Integration process"
+task Run-LocalCI -depends Clean-Solution, Run-Tests, Package-Local -description "Local continuous integration process"
+task ReBuild-Solution -depends Clean-Solution, Run-Tests -description "Rebuild the code, with testing"
+
+task Get-Info -description "Display runtime information" {
+	exec { dotnet --info }
+}
+
+task Run-Tests -depends Run-Restore, Build-Solution -description "Run unit tests" {
+	exec { dotnet test --configuration $configuration  --logger "trx;LogFileName=Testresults.xml" } -workingDirectory src/BulkWriter.Tests
+}
+
+task Build-Solution -depends Get-Info -description "Build the solution" {
+	if($suffix -eq "") {
```

Comment:
Is it not possible to set the version earlier? At a minimum, turn this into a variable so we don't repeat the $suffix =eq "" condition.

This condition check seems to be the only usage of $suffix..perhaps we can rename it and set it to a boolean value.

Reply:
I suggest getting rid of `$suffix` entirely, just use `$version` for the whole version everywhere.
Get rid of this if statement and just always set the project properties.
Make sure to add that Directory.Build.props file to your .gitignore though so it doesn't get committed each time.